### DISCUSSION TITLE
Does Sidekiq jobs run in the same queue concurrently or in parallel?

### DISCUSSION THREAD

**Discussion by pedrogusfer** [sidekiq/sidekiq#5477] [Q&A] [answered]
If Sidekiq uses multithreading, are the jobs running concurrently?

For example, I have two queues: **emails** and **payments**. Considering that I started one Sidekiq process, using `bundle exec sidekiq -c 2 -q emails, payments`. In that context, if I schedule one job for each queue, using `perform_async`, will these jobs run one after, concurrently, or in parallel, at the same time?

Must I run two processes, one process for each queue, if I want to run the jobs in parallel?

**Comment by mperham** [comment]
Concurrently. Jobs will run simultaneously.

PS Your cmd line is wrong: -q emails -q payments

**Reply by pedrogusfer** [reply]
If I want to run the jobs at the same time, in parallel, do I have to start two Sidekiq processes? Or is there another way to achieve that?

**Reply by himanish-k** [reply]
@pedroaugustofsilva Ever got an answer to this?

**Comment by mperham** [comment, marked as answer]
They run concurrently, on separate threads. Sidekiq is pure Ruby so it has any threading limitations of the underlying platform.

**Reply by pedrogusfer** [reply]
How does the Sidekiq run threads in parallel if Ruby's Global Interpreter Lock doesn't allow threads to run in parallel?

**Reply by beaucephus** [reply]
They run concurrently, on separate threads.
